### PR TITLE
Implement the method searchByPath in the class SearchVideo

### PR DESCRIPTION
**Pull request by TheYisusGithub**
In this commit I create the class ISearch and implement the method in the class SearchVideo

### CODE REVIEW

File: src/main/java/com/fundation/logic/Model/ISearch.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
+ /*
```
Comment (aburgosu) [inline src/main/java/com/fundation/logic/Model/ISearch.java]: Delete this class from the remote repository

File: src/main/java/com/fundation/logic/Model/SearchVideo.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  * ("Confidential Information"). You shall not
+  * disclose such Confidential Information and shall use it only in
+  * accordance with the terms of the license agreement you entered into
+  * with Jalasoft.
+  */
+ /* Implements the model class File and the getter and setter´s methods
+  * @author Jesus Menacho
+  * @version 1.0 *
+  */
+package com.fundation.logic.Model;
```
Comment (aburgosu) [inline src/main/java/com/fundation/logic/Model/SearchVideo.java]: Move the package and imports sentences up to author

File: src/main/java/com/fundation/logic/Model/SearchVideo.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+import java.util.List;
+import java.util.stream.Stream;
+
+public class SearchVideo implements ISearch {
+
+    public SearchVideo(){
+    }
+    public static List searchByPath(final String pathDirectory) {
+        List<String> listFileAndDirectory = new ArrayList<String>();
+        File file=new File(pathDirectory);
```
Comment (aburgosu) [inline src/main/java/com/fundation/logic/Model/SearchVideo.java]: space before and after "=" symbol

file=new <--

File: src/main/java/com/fundation/logic/Model/SearchVideo.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+        File[] allSubFiles=file.listFiles();
```
Comment (aburgosu) [inline src/main/java/com/fundation/logic/Model/SearchVideo.java]: space before and after "=" symbol
allSubFiles=file <--

File: src/main/java/com/fundation/logic/Model/SearchVideo.java [hunk 5]
```diff
@@ -0,0 +1,36 @@
+ /*
```
Comment (titor-oopart) [inline src/main/java/com/fundation/logic/Model/SearchVideo.java]: use /** for start all coment

File: src/main/java/com/fundation/logic/Model/SearchVideo.java [hunk 6]
```diff
@@ -34,8 +34,19 @@ public List searchAnyWant() {
         File file = new File(FILE_PATH);
         File[] allSubFiles=file.listFiles();
         for (File fileExtractor : allSubFiles) {
-            if (criteria.getPath().equals(""))
-            listFileAndDirectory.add(fileExtractor.getAbsolutePath());
+            //search by path
+            if (!(criteria.getPath().equals("")) && (criteria.getFileName().equals("")) &&(criteria.getExtension().equals("")) ) {
+                listFileAndDirectory.add(fileExtractor.getAbsolutePath());
+            }
+            if (!(criteria.getPath().equals("")) && (!criteria.getFileName().equals("")) &&(criteria.getExtension().equals("")) ) {
```
Comment (titor-oopart) [inline src/main/java/com/fundation/logic/Model/SearchVideo.java]: space before and after "&&"

File: src/main/java/com/fundation/logic/Model/SearchVideo.java [hunk 7]
```diff
@@ -34,8 +34,19 @@ public List searchAnyWant() {
...
+                if (criteria.getFileName().equals(fileExtractor.getName())) {
+                    listFileAndDirectory.add(fileExtractor.getName());
+                }
+            }
+            if (!(criteria.getPath().equals("")) && (!criteria.getFileName().equals("")) &&(!criteria.getExtension().equals("")) ){
```
Comment (titor-oopart) [inline src/main/java/com/fundation/logic/Model/SearchVideo.java]: Always  a space before this "{" at ends of your line

File: src/main/java/com/fundation/logic/Model/SearchVideo.java [hunk 8]
```diff
@@ -34,8 +34,19 @@ public List searchAnyWant() {
...
+                listFileAndDirectory.add(fileExtractor.getName().contains());
+            }
+
```
Comment (titor-oopart) [inline src/main/java/com/fundation/logic/Model/SearchVideo.java]: delete the empty line

File: src/main/java/com/fundation/logic/Model/SearchVideo.java [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Stream;
+
+public class SearchVideo implements ISearch {
+
+    public SearchVideo(){
```
Comment (titor-oopart) [inline src/main/java/com/fundation/logic/Model/SearchVideo.java]: missing a space before "{"

File: src/main/java/com/fundation/logic/Model/SearchVideo.java [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+package com.fundation.logic.Model;
+
+import java.io.File;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Stream;
+
```
Comment (titor-oopart) [inline src/main/java/com/fundation/logic/Model/SearchVideo.java:19]: missing the autor

File: gradle/wrapper/gradle-wrapper.properties [hunk 11]
```diff
@@ -0,0 +1,5 @@
+distributionBase=GRADLE_USER_HOME
+distributionPath=wrapper/dists
+distributionUrl=https\://services.gradle.org/distributions/gradle-4.10.3-bin.zip
+zipStoreBase=GRADLE_USER_HOME
+zipStorePath=wrapper/dists
```
Comment (MelissaRoman06) [inline gradle/wrapper/gradle-wrapper.properties]: Unnecessary file

File: gradlew [hunk 12]
```diff
@@ -0,0 +1,172 @@
...
+
+# Collect all arguments for the java command, following the shell quoting and substitution rules
+eval set -- $DEFAULT_JVM_OPTS $JAVA_OPTS $GRADLE_OPTS "\"-Dorg.gradle.appname=$APP_BASE_NAME\"" -classpath "\"$CLASSPATH\"" org.gradle.wrapper.GradleWrapperMain "$APP_ARGS"
+
+# by default we should be in the correct project dir, but when run from Finder on Mac, the cwd is wrong
+if [ "$(uname)" = "Darwin" ] && [ "$HOME" = "$PWD" ]; then
+  cd "$(dirname "$0")"
+fi
+
+exec "$JAVACMD" "$@"
```
Comment (MelissaRoman06) [inline gradlew]: Unnecessary file

File: gradlew.bat [hunk 13]
```diff
@@ -0,0 +1,84 @@
...
+:fail
+rem Set variable GRADLE_EXIT_CONSOLE if you need the _script_ return code instead of
+rem the _cmd.exe /c_ return code!
+if  not "" == "%GRADLE_EXIT_CONSOLE%" exit 1
+exit /b 1
+
+:mainEnd
+if "%OS%"=="Windows_NT" endlocal
+
+:omega
```
Comment (MelissaRoman06) [inline gradlew.bat]: Unnecessary file

File: src/main/java/com/fundation/logic/Model/SearchVideo.java [hunk 14]
```diff
@@ -0,0 +1,65 @@
...
+    public List searchByPath() {
+        if ((criteria.getPath().equals("")) && (criteria.getFileName().equals("")) && (criteria.getExtension().equals(""))) {
+            return null;
+        }
+        if ((criteria.getPath().equals(""))) {
+            return null;
+        }
+        final String FILE_PATH = criteria.getPath();
+        List<String> listFileAndDirectory = new ArrayList<String>();
```
Comment (MelissaRoman06) [inline src/main/java/com/fundation/logic/Model/SearchVideo.java]: For all search methods, I recommend to return a list containing File objects

File: src/main/java/com/fundation/logic/Model/SearchVideo.java [hunk 15]
```diff
@@ -0,0 +1,65 @@
...
+                    listFileAndDirectory.add(fileExtractor.getAbsolutePath());
+                }
+            }
+            if (!(criteria.getPath().equals("")) && (criteria.getFileName().equals("")) && (!criteria.getExtension().equals(""))) {
+                if (fileExtractor.getName().contains(criteria.getExtension())) {
+                    listFileAndDirectory.add(fileExtractor.getAbsolutePath());
+                }
+            }
+        }
+        return listFileAndDirectory;
```
Comment (MelissaRoman06) [inline src/main/java/com/fundation/logic/Model/SearchVideo.java:62]: For all search methods, I recommend to return a list containing File objects

**Pull request merged**